File: LeetCode Practice Problems/1038- Binary Search Tree to Greater Sum Tree.java
class Solution {
    int s = 0;
    void find(TreeNode root)
    {
        if(root == null)
            return;
        find(root.right);
        s = s + root.val;
        root.val = s;
        find(root.left);
    }
    public TreeNode bstToGst(TreeNode root) {
        find(root);
        return root;
    }
}

File: LeetCode/LeetCode Practice Problems/1038- Binary Search Tree to Greater Sum Tree.cpp
class Solution {
public:
    int solve(TreeNode* r,int val) {
        if(r != NULL) {
            int v = solve(r->right , val);
            if(v != 0) val = 0;
            r->val += v + val;
            int l = solve(r->left , r->val);
            return (l == 0)?r->val:l;
        }
        return 0;
    }
    
    TreeNode* bstToGst(TreeNode* root) {
        solve(root , 0);
        return root;
    }
};
